### DISCUSSION TITLE
Enter Key not working in Insert Mode

### DISCUSSION THREAD

**Discussion by DanielHidalgoChica** [JetBrains/ideavim#1000] [Q&A]
[ideavimrc.txt](https://github.com/user-attachments/files/17267333/ideavimrc.txt)
I have just installed CLion 2024.2.2 (in Ubuntu 22.04.1) with version 2.16.0 of the IdeaVim plugin. Enter Key is just not working in Insert Mode: it does nothing in the editor when pressed, but Ubuntu makes this "bup" sound (like when you press TAB when autocompleting commands in terminal). 

I can't find a reason why. My ~/.ideavim file is not the problem (I will attach it anyway) as it is an exact copy of my ~/.vimrc and Enter Key has normal behaviour in Vim, as it also works when the Plugin is not enabled. 

I have seen some other threads where people some years ago had similar issues, but I haven't found any solutions out there.

Please, how can I go about solving this? Any help would be much, much appreciated.

Thanks in advance.

**Comment by Miskho** [comment]
@DanielHidalgoChica did you manage to solve this? I am also experiencing the same problem.

**Comment by LunaCodeGirl** [comment]
Same issue here....

**Comment by OptimusCrime** [comment]
@DanielHidalgoChica : I tested your config file, and enter key did indeed stop to work in Insert mode. The problem is somewhere in your vimrc file (it is not compatible with ideavim). When I rolled back to my own configuration file, enter worked as expected in Insert mode again. 

You could try to remove parts of the config file until enter starts behaving as it should again.

@LunaCodeGirl : I suspect that you are having the same issue (unless your don't have a config file).

**Comment by lippfi** [comment]
```
inoremap <expr> <cr> 
   \   getline('.') =~ '^\s*{$'                  ? "<cr>}<esc>O" 
   \ : getline(".")[col(".")-2:col(".")-1]=="{}" ? "<cr><esc>O"
   \ :                                             "<cr>"
```
The `getline` is not implemented in IdeaVim

**Comment by listenerri** [comment]
If you have the IdeaVimExtension plugin installed, you can try disabling it to check if the problem is resolved, and if, like me, it is causing the problem, I fixed it as follows: 
```
disable the IdeaVimExtension plugin
restart the IDE
enable the IdeaVimExtension plugin
restart the IDE again
```
Or just reinstall IdeaVimExtension plugin
